data-prepper-core/src/main/java/org/opensearch/dataprepper/pipeline/Pipeline.java

```diff
@@ -255,9 +255,16 @@ private void shutdownExecutorService(final ExecutorService executorService, fina
     List<Future<Void>> publishToSinks(final Collection<Record> records) {
         final int sinksSize = sinks.size();
         final List<Future<Void>> sinkFutures = new ArrayList<>(sinksSize);
-        router.route(records, sinks, (sink, events) ->
-                sinkFutures.add(sinkExecutorService.submit(() -> sink.output(events), null))
+        final List<DataFlowComponent<Sink>> firstSink = sinks.subList(0,1);
```

Comment:
This picks the first sink to have a copy. What if the first sink does not need the copy?

```
sink:
  - opensearch:
  - pipeline:
  - pipeline:
```

In that example, it will perform the copies anyway. So there is no performance gain.

Also, by using the first sink, you may also perform the copy if conditionally routing to the second sink. Consider the example below.

```
  sink:
    - pipeline:
        routes: [route-a]
    - pipeline:
        routes: [route-b]
```

This will perform the copy for all events which go to route-b even though they don't need to be copied for the performance gain.